Consistent interface

soniacq opened the pull request:
Merging master with consistent_interface branch.

brittainhard commented:
@yamsgithub this is ready for review.

yamsgithub commented:
Comments on the dashboard:

1. The dashboard was supposed to be accessed from a tab on the menu bar.
2. The domain table has only 10 entries while there definitely should be more. Are you displaying only the top 10?
3. The data fetched by time should be plotted by domain
4. Currently we are not able to custom select the domains to plot
5. It does not seem to be able to handle larger datasets of 1000's of documents. There seems to be a request time out on the elasticsearch search query. This can be fixed by passing a request_timeout=300 (in seconds for example) to the es.search query.

brittainhard commented:
Thanks @yamsgithub , i'll give my comments on each of these.

1) This is a simple fix, and I can push up a commit immediately that will address this concern.
2) How many items do you want to show per table? How many items do you want to show on the plot?
3 & 4) I think that these things are out of scope for this part of the project. We will have to discuss this at the meeting today.
5. Are you getting specific bugs with timeouts? Is the page failing to load? Is it simply taking too long to load? Some clarification is needed here.

yamsgithub commented:
2. All the domains should be listed in the table.The plot can show the top 10
3. & 4.  What part of the project are we talking about. Cross filtering data for the plot is a mandatory requirement for the dashboard.
5. I am getting the following error when the data is quite large.

500 Internal Server Error

The server encountered an unexpected condition which prevented it from fulfilling the request.

Traceback (most recent call last):
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/media/data/yamuna/Memex/test/domain_discovery_tool/vis/server.py", line 287, in statistics
    pages_dates = self._crawler.getPagesDates(session)
  File "/media/data/yamuna/Memex/test/domain_discovery_tool/vis/crawler_model_adapter.py", line 180, in getPagesDates
    return self._crawlerModel.getPagesDates(session)
  File "/media/data/yamuna/Memex/test/domain_discovery_tool/models/crawlermodel.py", line 933, in getPagesDates
    return get_pages_datetimes(es_info["activeCrawlerIndex"])
  File "/media/data/yamuna/Memex/test/domain_discovery_tool/elastic/get_documents.py", line 200, in get_pages_datetimes
    items = es.search(index_name, size=100000)["hits"]["hits"]
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/elasticsearch-1.6.0-py2.7.egg/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/elasticsearch-1.6.0-py2.7.egg/elasticsearch/client/__init__.py", line 506, in search
    params=params, body=body)
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/elasticsearch-1.6.0-py2.7.egg/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/media/data/yamuna/miniconda2/envs/ddt/lib/python2.7/site-packages/elasticsearch-1.6.0-py2.7.egg/elasticsearch/connection/http_urllib3.py", line 82, in perform_request
    raise ConnectionTimeout('TIMEOUT', str(e), e)
ConnectionTimeout: ConnectionTimeout caused by - ReadTimeoutError(HTTPConnectionPool(host='localhost', port=9200): Read timed out. (read timeout=10))

brittainhard commented:
@yamsgithub this is ready for another look.

yamsgithub commented:
The changes look fine. I will merge the PR

soniacq committed:
Making a responsive 'search' textfield which is used to filter by terms.

soniacq committed:
Making a responsive 'search' textfield which is used to filter by terms.

soniacq committed:
Making a responsive 'search' textfield which is used to filter by terms.

soniacq committed:
Making a responsive 'search' textfield which is used to filter by terms.

soniacq committed:
Removed searchicon.png because it has an incorrect orientation. Making consistent the orientation on 'search' icons.

soniacq committed:
Removed searchicon.png because it has an incorrect orientation. Making consistent the orientation on 'search' icons.

soniacq committed:
Removed searchicon.png because it has an incorrect orientation. Making consistent the orientation on 'search' icons.

soniacq committed:
Removed searchicon.png because it has an incorrect orientation. Making consistent the orientation on 'search' icons.

soniacq committed:
Changed the 'search' image used to create chips after applying a filter (on the View component).

soniacq committed:
Changed the 'search' image used to create chips after applying a filter (on the View component).

soniacq committed:
Changed the 'search' image used to create chips after applying a filter (on the View component).

soniacq committed:
Changed the 'search' image used to create chips after applying a filter (on the View component).

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Changed icon buttons with label buttons on the Header in order to be more explicit the button functionality. Replaced 'change domain icon' with a button which has the 'Switch domain' label. Replaced 'monitoring icon' with a button which has the 'Monitoring' label.

soniacq committed:
Merge branch 'consistent_interface' of https://github.com/ViDA-NYU/domain_discovery_tool into consistent_interface

soniacq committed:
Merge issue. Fixed the name of a function.

soniacq committed:
Merge issue. Fixed the name of a function.

soniacq committed:
Merge issue. Fixed the name of a function.

soniacq committed:
Merge issue. Fixed the name of a function.

soniacq committed:
Making consistent design of buttons in SearchTabs component.

soniacq committed:
Making consistent design of buttons in SearchTabs component.

soniacq committed:
Making consistent design of buttons in SearchTabs component.

soniacq committed:
Setting up the backgroundColor color button on the Header (for 'Switch domain' and 'Monitoring' buttons.)

soniacq committed:
Setting up the backgroundColor color button on the Header (for 'Switch domain' and 'Monitoring' buttons.)

soniacq committed:
Setting up the backgroundColor color button on the Header (for 'Switch domain' and 'Monitoring' buttons.)

soniacq committed:
Making consistent design of buttons in DeepCrawling component.

soniacq committed:
Making consistent design of buttons in DeepCrawling component.

soniacq committed:
Making consistent design of buttons in DeepCrawling component.

soniacq committed:
Making consistent design of buttons in FocusedCrawling component.

soniacq committed:
Making consistent design of buttons in FocusedCrawling component.

soniacq committed:
Making consistent design of buttons in FocusedCrawling component.

soniacq committed:
Fixed issue #12: The delete filter list should be scrollable.

soniacq committed:
Fixed issue #12: The delete filter list should be scrollable.

soniacq committed:
Fixed issue #12: The delete filter list should be scrollable.

soniacq committed:
Fixed issue #12: The delete filter list should be scrollable.

nandukalidindi committed:
Rename forward/backward crawl buttons on the multi select popup

soniacq committed:
Fixed issue #31: tagging pages with the filter 'neutral'. When the user uses 'Tag All' (for example 'Neutral' tag) and all snippets desapear, instead of show an empty page, ddt will display snippets for pages that are still neutral.

nandukalidindi committed:
Add download button on the popup modal during model export

Opening a new window tab inside an AJAX request turned out
to be almost impossible without using some non-reliable hacks.
Hence providing a button on the popup modal for user to click.

soniacq committed:
Fixed issue #31: Fixed issue related with custom tagging pages of Neutral pages.

soniacq committed:
Removed unnecessary import in View component.

soniacq committed:
Fixed issue #58: Pagination is incorrect after applying some filter.

soniacq committed:
Fixed issue #37: file name and 'Choose File' misaligned. Changed the position of custom tagging.

soniacq committed:
Fixed issue #37 for 'Run multiple queries': file name and 'Choose File' misaligned.

soniacq committed:
Fixed issue #37 for 'Deep crawling': file name and 'Choose File' misaligned.

soniacq committed:
Merged Header.

soniacq committed:
Improving components style into searchTab.

soniacq committed:
Merge branch 'consistent_interface' of https://github.com/ViDA-NYU/domain_discovery_tool into consistent_interface

soniacq committed:
Changed DropDownMenu style to be consistent with the whole ddt interface.

soniacq committed:
Changed DropDownMenu style to be consistent with the whole ddt interface.

soniacq committed:
Aligning elements on the View to create order and organisation among elements. Standardised style in texts.

soniacq committed:
Aligning elements on the View to create order and organisation among elements. Standardised style in texts.

soniacq committed:
Using label buttons instead of just icon button to adding or reload terms.

soniacq committed:
Using label buttons instead of just icon button to adding or reload terms.

soniacq committed:
Making consistent design of Add and Remove domain buttons. Also, label buttons were used instead of just icon buttons.

soniacq committed:
Making consistent design of Add and Remove domain buttons. Also, label buttons were used instead of just icon buttons.

soniacq committed:
Merging Header.

soniacq committed:
Merging Header.

soniacq committed:
Merging Header.

soniacq committed:
Merging Home.

soniacq committed:
Merging SearchTabs.

soniacq committed:
Merging Header.

soniacq committed:
Merging Header.

soniacq committed:
Merging SearchTabs.

soniacq committed:
Merging SearchTabs.

soniacq committed:
Fixing merge Home.
